ts/addon.ts

```diff
@@ -97,6 +106,47 @@ export default addon({
     return !['in-repo-a', 'in-repo-b'].includes(addon.name);
   },
 
+  _registerCollisionDetectionPreprocessor(registry: PreprocessRegistry) {
+    registry.add('js', {
+      name: 'ember-cli-typescript-collision-check',
+      toTree: (input, path) => {
+        if (path !== '/') return input;
+
+        let addon = this;
+        let checked = false;
+        let stew = require('broccoli-stew') as typeof import('broccoli-stew');
+
+        return stew.afterBuild(input, function() {
+          if (!checked) {
+            checked = true;
+            addon._checkForFileCollisions(this.inputPaths[0]);
```

Comment:
Just checking: what invariant guarantees that `this.inputPaths` always has at least one element?

Reply:
Broadly speaking, Broccoli plugins have one element in `inputPaths` per node they receive in their constructor.

In the case of `afterBuild` (and `Funnel` and a lot of other common plugins), it accepts a single node and then passes it in a singleton array when it calls `super` in its constructor, so `inputPaths[0]` in this case is the path on disk for the `input` tree we pass in a couple lines above.

Reply:
Excellent; thank you! TIL!